pkg/sqlite/image.go

```diff
@@ -141,12 +150,60 @@ func (i *ImageLoader) loadManifests(manifests string, annotationsFile *registry.
 		return fmt.Errorf("error getting csv from bundle %s: %s", bundle.Name, err)
 	}
 
+	// before generating synthetic package manifest file and loading bundle - check to see if bundles provided
+	// to add command are related - if so bundle commutativity must be ensured. we get this for free with semver based bundles.
+
+	// get array of all bundle types
+	// check if related
+	// if so order them properly
+	// if not, or only 1 , do nothing
+	bdata := bundleData{
+		annotationsFile: annotationsFile,
+		bcsv:            bcsv,
+		bundle:          bundle,
+	}
+	addedBundles = append(addedBundles, bdata)
+
+	var sortedBundles []bundleData
+	if i.bundles > 0 {
+		// we are still cycling through the provided bundles - return early
+		return nil
+	} else {
+		// there are no bundles left provided to the add arg
+
+		// check to see if bundle types are related
+		// TODO do we have code that does this already somewhere?
+		sortedBundles = checkRelatedBundles(addedBundles)
+
+		// clear array
+		addedBundles = nil
```

Comment:
bug: this needs to be `deferred` to ensure it happens on every error exit in the function context